Bottom: 6ec7010c589e53310d4e9ffc5a91ba911210bf4d
Top:    7dc2e83797b5b3d45c42b1ec23c6eb4ab8451e3e
Author: Ian Kent <raven@themaw.net>
Date:   2013-08-19 15:17:41 +0800

Refresh of autofs-5.0.7-allow-use-of-hosts-map-in-maps.patch

---

diff --git a/modules/parse_sun.c b/modules/parse_sun.c
index 9db9e18..f6e79b7 100644
--- a/modules/parse_sun.c
+++ b/modules/parse_sun.c
@@ -1015,7 +1015,7 @@ static int parse_mapent(const char *ent, char *g_options, char **options, char *
 		}
 
 		if (!validate_location(logopt, ent_chunk)) {
-			error(LOGOPT_ANY, "revalidate_location failed");
+			error(LOGOPT_ANY, "validate_location failed");
 			free(ent_chunk);
 			free(myoptions);
 			free(loc);
@@ -1442,8 +1442,9 @@ int parse_mount(struct autofs_point *ap, const char *name,
 			error(LOGOPT_ANY, "options %s", options);
 
 			l = parse_mapent(p, options, &myoptions, &loc, ap->logopt);
-			if (!(strstr(myoptions, "fstype=autofs") &&
-			     strstr(myoptions, "hosts")) || !l) {
+			if (!l &&
+			    !(strstr(options, "fstype=autofs") &&
+			      strstr(options, "hosts"))) {
 				error(LOGOPT_ANY, "I think I'm a hosts map? l %d", l);
 				cache_delete_offset_list(mc, name);
 				cache_multi_unlock(me);
